ibm/resource_ibm_cis_healthcheck.go

```diff
@@ -2,331 +2,364 @@ package ibm
+		opt.SetDescription(description.(string))
 	}
-	if method, ok := d.GetOk("method"); ok {
-		monitorNew.Method = method.(string)
+	if method, ok := d.GetOk(cisGLBHealthCheckMethod); ok {
+		opt.SetMethod(method.(string))
 	}
-	if timeout, ok := d.GetOk("timeout"); ok {
-		monitorNew.Timeout = timeout.(int)
+	if timeout, ok := d.GetOk(cisGLBHealthCheckTimeout); ok {
+		opt.SetTimeout(int64(timeout.(int)))
 	}
-	if retries, ok := d.GetOk("retries"); ok {
-		monitorNew.Retries = retries.(int)
+	if retries, ok := d.GetOk(cisGLBHealthCheckRetries); ok {
+		opt.SetRetries(int64(retries.(int)))
 	}
-	if interval, ok := d.GetOk("interval"); ok {
-		monitorNew.Interval = interval.(int)
+	if interval, ok := d.GetOk(cisGLBHealthCheckInterval); ok {
+		opt.SetInterval(int64(interval.(int)))
 	}
-	if follow_redirects, ok := d.GetOk("follow_redirects"); ok {
-		monitorNew.FollowRedirects = follow_redirects.(bool)
+	if followRedirects, ok := d.GetOk(cisGLBHealthCheckFollowRedirects); ok {
+		opt.SetFollowRedirects(followRedirects.(bool))
 	}
-	if allow_insecure, ok := d.GetOk("allow_insecure"); ok {
-		monitorNew.AllowInsecure = allow_insecure.(bool)
+	if allowInsecure, ok := d.GetOk(cisGLBHealthCheckAllowInsecure); ok {
+		opt.SetAllowInsecure(allowInsecure.(bool))
 	}
-	if port, ok := d.GetOk("port"); ok {
-		monitorNew.Port = port.(int)
+	if port, ok := d.GetOk(cisGLBHealthCheckPort); ok {
+		opt.SetPort(int64(port.(int)))
 	}
-	var monitor *v1.Monitor
-	var monitorObj v1.Monitor
 
-	monitor, err = cisClient.Monitors().CreateMonitor(cisId, monitorNew)
+	result, resp, err := sess.CreateLoadBalancerMonitor(opt)
 	if err != nil {
-		log.Printf("CreateMonitors Failed %s\n", err)
+		log.Printf("create global load balancer health check failed %s", resp)
 		return err
 	}
-	monitorObj = *monitor
-	//Set unique TF Id from concatenated CIS Ids
-	d.SetId(monitorObj.Id + ":" + cisId)
-	d.Set("path", monitorObj.Path)
-
-	return resourceCIShealthCheckRead(d, meta)
+	log.Printf("global load balancer created successfully : %s", *result.Result.ID)
+	d.SetId(convertCisToTfTwoVar(*result.Result.ID, crn))
+	return resourceCISHealthCheckRead(d, meta)
 }
 
-func resourceCIShealthCheckRead(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISHealthCheckRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
 	if err != nil {
 		return err
 	}
-	monitorId, cisId, err := convertTftoCisTwoVar(d.Id())
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	var monitor *v1.Monitor
-	monitor, err = cisClient.Monitors().GetMonitor(cisId, monitorId)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewGetLoadBalancerMonitorOptions(monitorID)
+
+	result, resp, err := sess.GetLoadBalancerMonitor(opt)
 	if err != nil {
-		if checkCisMonitorDeleted(d, meta, err, monitor) {
```

Comment:
Please check if monitor is deleted
you can refer rate limit resource [here](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_cis_rate_limit.go#L413)

Reply:
checkCisMonitorDeleted() is a dead code since I have added Exists() method and returned nil there

Reply:
please check for response not nil 
please refer existing resource [here](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/resource_ibm_api_gateway_endpoint.go#L186)

Reply:
we have Exists() method which does what you are trying to say here. Thats why I am saying checkDelete() method is a dead code here. it Never executes till the Exists() method gets removed..
please understand it.I tested it multiple time with different scenario.
 other wise test my code yourself. if you are saying that checkDelete() method still executed on particular scenario, please let me know the scenario.

Reply:
okay..